Src/Core/Collections/CollectionAssertions.cs

```diff
@@ -209,65 +207,33 @@ public AndConstraint<TAssertions> Equal(IEnumerable expected, string because = "
         protected void AssertSubjectEquality<TActual, TExpected>(IEnumerable expectation, Func<TActual, TExpected, bool> predicate,
             string because = "", params object[] reasonArgs)
         {
-            AssertionScope assertion = Execute.Assertion.BecauseOf(because, reasonArgs);
-
-            bool subjectIsNull = ReferenceEquals(Subject, null);
-            bool expectationIsNull = ReferenceEquals(expectation, null);
 
+            var subjectIsNull = ReferenceEquals(Subject, null);
```

Comment:
The one with the `Cast<TExpected>().ToArray()` is an example where the type is explicitly stated. In all other cases, I prefer not using var at all.